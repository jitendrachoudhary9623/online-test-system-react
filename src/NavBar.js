import React,{Component} from 'react';
class NavBar extends Component{

    constructor(props){
      super(props);
    }

    
    render(){
        return (<div className="grey darken-1 navbar-fixed ">
        <nav>
          <div className="nav-wrapper">
            <span className="brand-logo pad-2 left">{this.props.title}</span>
            <ul className="right">
            </ul>
          </div>
        </nav>
      </div>
            )
    }
}

NavBar.defaultProps={
  title:"CDK Test"
}
export default NavBar;