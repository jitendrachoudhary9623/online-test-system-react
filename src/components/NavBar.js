import React,{Component} from 'react';
class NavBar extends Component{
    constructor(props){
      super(props);
    }    
    
    render(){
        return (<div className="grey navbar-fixed ">
        <nav>
          <div className="nav-wrapper">
            <span className="brand-logo pad-2 left blue-text text-darken-2" style={{marginLeft:30}}>{this.props.title}</span>
            <ul className="right">
            </ul>
          </div>
        </nav>
      </div>
            )
    }
}

NavBar.defaultProps={
  title:"CDK Test"
}
export default NavBar;