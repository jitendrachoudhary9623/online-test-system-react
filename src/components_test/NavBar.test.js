import React, { Component } from 'react';
import NavBar from "../NavBar"
import { shallow } from 'enzyme'

describe('UI Test for Nav Bar', () => {

    it('Nav Bar to be defined', () => {
        expect(NavBar).toBeDefined();
    });

    it('Nav Bar should render correctly', () => {
    
        const snapshot = shallow(
            <NavBar title="CDK Test"/>
            );
        expect(snapshot).toMatchSnapshot();
      });

      it('Nav Bar having proper Title as specified in the props', () => {
        var title="Title";
        const snapshot = shallow(
            <NavBar title={title}/>
            );
            expect(snapshot.find('span').text()).toEqual(title);
      });
})