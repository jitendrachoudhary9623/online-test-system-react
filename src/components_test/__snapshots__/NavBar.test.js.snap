exports[`UI Test for Nav Bar Nav Bar should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar
    title="CDK Test" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav>
        <div
                className="nav-wrapper">
                <span
                        className="brand-logo pad-2 left blue-text text-darken-2"
                        style={
                                Object {
                                        "marginLeft": 30,
                                      }
                        }>
                        CDK Test
                </span>
                <ul
                        className="right" />
        </div>
</nav>,
      "className": "grey navbar-fixed ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="nav-wrapper">
          <span
                    className="brand-logo pad-2 left blue-text text-darken-2"
                    style={
                              Object {
                                        "marginLeft": 30,
                                      }
                    }>
                    CDK Test
          </span>
          <ul
                    className="right" />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="brand-logo pad-2 left blue-text text-darken-2"
              style={
                            Object {
                                          "marginLeft": 30,
                                        }
              }>
              CDK Test
</span>,
            <ul
              className="right" />,
          ],
          "className": "nav-wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "CDK Test",
              "className": "brand-logo pad-2 left blue-text text-darken-2",
              "style": Object {
                "marginLeft": 30,
              },
            },
            "ref": null,
            "rendered": "CDK Test",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "right",
            },
            "ref": null,
            "rendered": null,
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": "nav",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav>
          <div
                    className="nav-wrapper">
                    <span
                              className="brand-logo pad-2 left blue-text text-darken-2"
                              style={
                                        Object {
                                                  "marginLeft": 30,
                                                }
                              }>
                              CDK Test
                    </span>
                    <ul
                              className="right" />
          </div>
</nav>,
        "className": "grey navbar-fixed ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="nav-wrapper">
            <span
                        className="brand-logo pad-2 left blue-text text-darken-2"
                        style={
                                    Object {
                                                "marginLeft": 30,
                                              }
                        }>
                        CDK Test
            </span>
            <ul
                        className="right" />
</div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="brand-logo pad-2 left blue-text text-darken-2"
                style={
                                Object {
                                                "marginLeft": 30,
                                              }
                }>
                CDK Test
</span>,
              <ul
                className="right" />,
            ],
            "className": "nav-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "CDK Test",
                "className": "brand-logo pad-2 left blue-text text-darken-2",
                "style": Object {
                  "marginLeft": 30,
                },
              },
              "ref": null,
              "rendered": "CDK Test",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "right",
              },
              "ref": null,
              "rendered": null,
              "type": "ul",
            },
          ],
          "type": "div",
        },
        "type": "nav",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
